R/model.R

```diff
@@ -1387,14 +1306,50 @@ diagnose_method <- function(data = NULL,
     output_dir = output_dir,
     output_basename = output_basename
   )
-  cmdstan_procs <- CmdStanProcs$new(
-    num_procs = 1,
-    show_stdout_messages = FALSE,
-    show_stderr_messages = TRUE
-  )
-  runset <- CmdStanRun$new(args = cmdstan_args, procs = cmdstan_procs)
+  runset <- CmdStanRun$new(args, procs)
   runset$run_cmdstan()
 
   CmdStanDiagnose$new(runset)
 }
 CmdStanModel$set("public", name = "diagnose", value = diagnose_method)
+
+
+
+# internal ----------------------------------------------------------------
+
+assert_valid_opencl <- function(opencl_ids, cpp_options) {
+  if (is.null(cpp_options[["stan_opencl"]])
+      && !is.null(opencl_ids)) {
+    stop("'opencl_ids' is set but the model was not compiled with for use with OpenCL.",
+         "\nRecompile the model with 'cpp_options = list(stan_opencl = TRUE)'",
+         call. = FALSE)
+  }
+  invisible(opencl_ids)
+}
```

Comment:
return first argument invisibly now so change from `check_` to `assert_` (following naming conventions in `checkmate`)